File: python/day68.py
import hashlib
import getpass

# 1. Get hidden password input from the user
password = getpass.getpass("Enter your password: ")

# 2. Convert the password string to bytes and generate the MD5 hash
md5_hash = hashlib.md5(password.encode('utf-8')).hexdigest()

# 3. Save the hash to a text file
filename = "password_hash.txt"
with open(filename, "w") as file:
    file.write(md5_hash)

print(f"Success! The MD5 hash has been saved to '{filename}'.")
print(f"Hash: {md5_hash}")

File: js/day68.js
const crypto = require('crypto');
const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter password: ', (password) => {
    // 1. Generate MD5 hash of the password
    const hash = crypto.createHash('md5').update(password).digest('hex');

    // 2. Save the hash to a txt file
    fs.writeFile('password_hash.txt', hash, (err) => {
        if (err) throw err;
        console.log('MD5 hash successfully saved to password_hash.txt');
    });

    rl.close();
});
